### PR TITLE
Partial fixes

### PR DESCRIPTION
**Pull request by ryanbreen**
Remove npm shrinkwrap and expose proxies from proxy_manager by file name, not by path.

### CODE REVIEW

**Comment by ryanbreen** [comment]
Just a whole mess of fixes necessitated by updating the deps.  Lots of changes in node-http-proxy, multer, and body-parser.

All tests should pass now, but I'm not 100% happy with the urlencoded bodyparser charset thing.  I worry that'll lead to random failures in client apps.

File: lib/proxy_manager.js [hunk 1]
```diff
@@ -167,15 +167,15 @@ function loadConfigFiles(cb) {
 
     // Require a '.json' file ending and reject dotfiles
     var config_files = [];
-    for (var i in files) {
-      if (path.basename(files[i])[0] === '.') {
-        continue;
-      } else if (!/^.*\.(json)$/i.test(files[i])) {
-        continue;
+    files.forEach(function(file) {
```
Comment (theopak) [inline lib/proxy_manager.js:170]: :+1:

File: lib/proxy_manager.js [hunk 2]
```diff
@@ -167,15 +167,15 @@ function loadConfigFiles(cb) {
...
+      if (path.basename(file)[0] === '.') {
+        return;
+      } else if (!/^.*\.(json)$/i.test(file)) {
+        return;
       } else {
-        config_files.push(path.resolve(CONFIG_DIR, files[i]));
+        config_files.push(file);
```
Comment (theopak) [inline lib/proxy_manager.js:176]: Ok.

Comment (ryanbreen) [inline lib/proxy_manager.js:176]: The bootstrap tests rely on file paths having a certain form.  Feel free to change to resolve, but the fix needs to keep the bootstrap tests working.  I took the easy way out.

**Comment by theopak** [comment]
LGTM!

Thanks for fixing up the parts of this that I couldn't figure out.

**Pull request merged**